### PR TITLE
Refactoring of the Kolasudualversions branch to support loading v1 parser traces

### PR DESCRIPTION
**Pull request by alessiostalla**
Now it works. I've taken the liberty to move some stuff around.
However, I've had to support two slightly different EMF/JSON formats.
One contains absolute references such as:
```
"eClass" : "https://strumenta.com/rpg/model#//CompilationUnit",
```
The other one contains relative references such as:
```
"eClass":"#/0/CompilationUnit"
```

And we have to handle them differently.

The first format (that is found for example in the rpg-to-java transpilation trace) requires that the metamodel is stored in other resources in the same resource set as the model resource.
The second format (that is found for example in the RPG ast's) requires that the metamodel be part of the same resource as the model.

I think we should eventually settle on a single format (IMHO the first is better as it doesn't require shuffling EPackages around when loading, but it's more verbose) and understand why we're generating two different formats.

### CODE REVIEW

**Comment by ftomassetti** [comment]
I think there is somewhere an option about including or not the metamodel of the languages in the resource generated

File: src/interop/ecore.ts [hunk 1]
```diff
@@ -761,12 +509,12 @@ class ReferencesTracker {
  * @param data the input string or object.
  * @param resource where to look for to resolve references to types.
  */
-export function loadEObject(data: any, resource: Resource): EObject | undefined {
+export function loadEObject(data: any, resource: Resource, eClass?: EClass): EObject | undefined {
```
Comment (ftomassetti) [inline src/interop/ecore.ts]: Should we use unknown instead of any?

File: src/interop/parser-bench.ts [hunk 2]
```diff
@@ -30,4 +37,233 @@ export function saveForParserBench<R extends Node>(
...
+    }
+
+    getRole(): string {
+        return this.eo.eContainingFeature.get("name");
+    }
+
+    getPosition(): Position | null {
+        const raw = this.eo.get("position");
+        if (!raw) {
+            return raw;
```
Comment (ftomassetti) [inline src/interop/parser-bench.ts]: Should this be `return null`?

**Review by ftomassetti — APPROVED** [review]
Looks great, thank you!

**Comment by alessiostalla** [comment]
> I think there is somewhere an option about including or not the metamodel of the languages in the resource generated

As far as I understand it, the metamodel is not included in any of the two cases, but in one case, it looks like it was included then removed, in the other case, it looks like it was in a separate resource in the same ResourceSet. Probably we're using two different methods for generating these examples.

**Pull request merged**